src/dataobject.cpp

```diff
@@ -633,14 +634,13 @@ arma::mat dataobject::get_etas(const int n_eta, const bool strict,
   
   arma::mat ans(Uid.size(), n_eta); 
   for(size_t i = 0; i < Uid.size(); ++i) {
-    int start_row = Startrow[i];
+    int start_row = this->start(i);
     for(int j = 0; j < n_eta; ++j) {
-      if(eta_location[j] > 0) {
+      if(eta_location[j] >= 0) {
```

Comment:
I was confused at why this would be necessary for the `idata` change.  I reverted it (on top of c64ac9ab) and tests still pass.

Looking more closely at the existing upstream code, `eta_location` is the result of `std::find` on the eta labels, so 0-based.  By not considering `j = 0`, it mishandles an eta column if it's the first column.  So this is relevant for `data` too and is an independent fix for bug from gh-1037.  Okay.

Fwiw here's a regression test that fails without the `>` to `>=` equal change:

```R
test_that("etasrc works with ETA in first column", {
  mod <- param(mod, mode = 0)

  data <- expand.ev(amt = 100, ID = seq(4), cmt = 1)
  data <- mutate(data, ETA1 = rev(ID) / 10)
  data <- expand_observations(data, times = seq(5))
  data <- mutate(data, cmt = 0)

  set.seed(9812)
  expect_identical(
    mrgsim(mod, data, etasrc = "data"),
    mrgsim(mod, select(data, "ETA1", everything()), etasrc = "data")
  )
})
```

Reply:
Thanks; yes, I discovered this when creating the tests for this PR. I should have created another PR or at least written it up in the message. I'm going to revert this change and make it clean in a new PR in case this comes up in the future.